pgmpy/base/MAG.py

```diff
@@ -0,0 +1,295 @@
+        True
+        """
+        mark_uc_at_c = self.edges[u, c]["marks"][c]
+        mark_cv_at_c = self.edges[c, v]["marks"][c]
+
+        return mark_uc_at_c == ">" and mark_cv_at_c == ">"
+
+    def has_inducing_path(self, u, v, W):
+        """
+        Check if there exists an inducing path between two nodes relative to W.
+
+        An inducing path between u and v is a path such that:
+        - The path has length > 2 (at least one intermediate node),
+        - Every intermediate node is a collider on the path,
+        - Every intermediate node is either:
+            * in W, or
+            * an ancestor of u or v.
+
+        Parameters
+        ----------
+        u : Hashable
+            Source node.
+
+        v : Hashable
+            Target node.
+
+        W : set
+            Subset of nodes to check inducing paths through (often latents).
+
+        Returns
+        -------
+        bool
+            True if there exists an inducing path, False otherwise.
+
+        Examples
+        --------
+        >>> from pgmpy.base import MAG
+        >>> mag = MAG()
+        >>> mag.add_edge("X", "L", "-", ">")
+        >>> mag.add_edge("Y", "L", "-", ">")
+        >>> mag.latents = {"L"}
+        >>> mag.has_inducing_path("X", "Y", mag.latents)
+        True
+        """
+
+        is_inducing = True
+        for path in nx.all_simple_paths(self, source=u, target=v):
+            if len(path) <= 2:
+                continue
+
+            for i in range(1, len(path) - 1):
+                prev_node, curr_node, next_node = path[i - 1], path[i], path[i + 1]
+
+                if not self._is_collider(prev_node, curr_node, next_node):
+                    is_inducing = False
+                    break
+
+                ancestors_uv_vu = self.get_ancestors(u).union(self.get_ancestors(v))
+                if curr_node not in W and curr_node not in ancestors_uv_vu:
+                    is_inducing = False
+                    break
+
+        return is_inducing
+
+    def is_visible_edge(self, u, v) -> bool:
+        """
+        Check if an edge is visible.
+
+        An edge is visible if it exists and is not shielded by an inducing path
+        through latent variables.
+
+        Parameters
+        ----------
+        u : Hashable
+            First node.
+
+        v : Hashable
+            Second node.
+
+        Returns
+        -------
+        bool
+            True if the edge is visible, False otherwise.
+
+        Examples
+        --------
+        >>> from pgmpy.base import MAG
+        >>> mag = MAG()
+        >>> mag.add_edge("X", "Y", "-", ">")
+        >>> mag.is_visible_edge("X", "Y")
+        True
+        """
+        if not self.has_edge(u, v):
+            return False
+
+        graph_without_edge = self.copy()
+        graph_without_edge.remove_edge(u, v)
+        return not graph_without_edge.has_inducing_path(u, v, self.latents)
+
+    def lower_manipulation(self, X):
```

Comment:
This should accept an `inplace` boolean argument. Search for examples in the current codebase.